lib/number_format.js

```diff
@@ -150,16 +150,23 @@ var NumberFormat = React.createClass({
     this.onChangeHandler(e, this.props.onInput);
   },
   render: function render() {
-    var props = this.props;
+    var props = _extends({}, {}, this.props);
```

Comment:
Why double objects. It can be just 
```js
var props = _extends({},  this.props);
```

Also we can minimze by code by 
```js
['thousandSeperator', 'suffix', 'prefix', /* 'all others custom props' */].forEach((key) => {
 delete props[key];
});
```